File: library/json-mapper2/src/main/scala/com/anjunar/scala/schema/model/Link.scala
package com.anjunar.scala.schema.model

import com.anjunar.scala.mapper.annotations.{PropertyDescriptor, IgnoreFilter}

import scala.annotation.meta.field

@IgnoreFilter
case class Link(@(PropertyDescriptor @field)(title = "URL") url: String,
                @(PropertyDescriptor @field)(title = "Method") method: String,
                @(PropertyDescriptor @field)(title = "Relation") rel: String,
                @(PropertyDescriptor @field)(title = "Title") title: String)

File: application/src/main/javascript/workspace/projects/shared/src/lib/domain/container/Link.ts
import Entity from "../../mapper/annotations/Entity";
import Primitive from "../../mapper/annotations/Primitive";
import ActiveObject from "./ActiveObject";

@Entity("Link")
export default class Link {

    $type = "Link"

    @Primitive()
    url : string

    @Primitive()
    method : string

    @Primitive()
    rel : string

    @Primitive()
    title : string

    @Primitive()
    linkType : string

    @Primitive()
    body : ActiveObject

}